Music Update

SohanShet opened the pull request:
<<<Music Update>>>
theme music added
audio on/off button
music restarts when clicking StartGame
music stops on GameOver 
Sound Added: GameOver , RightInput , WrongInput
StartScreen again comes on Restarting

SohanShet committed:
MusicCommit:Music and Sounds added

Shriteen committed:
Refactor of music system

- Sound icon is put as background image rather than image element to allow auto switching using css
- Use variables to access audio elements instead of directly using ID
- Proper indentation and remove unnecessary code
- Remove MusicBit variable and associated code as we do not need to explicitely keep track of music state
- Music will start when game starts unless sound has been explicitely turned off
- Move toggling of sound button state from inline function to a separate function
- Change position of sound icon to bottom left to avoid overlap with gameplay

Shriteen committed:
Automatically focus typing field if music button is clicked during gameplay
- Improved UX

Shriteen commented:
I've made some updates to make it more robust. Do not merge yet as I'll try to integrate multiple music.

Shriteen committed:
Autoplay music if user has given permission

Shriteen committed:
Replace Music files with lighter versions

- Less storage as well as lower loading time

Shriteen committed:
Multiple music tracks
- Move variables declaration
- Play random songs from list
- When song is over, start another random song

Shriteen commented:
I've added playing from multiple songs randomly; If we wish to add more songs,  just put it inside assets/music/ and update the musicTracks array with their names

More things to do before merging
- The sound effect of correct word doesn't fit well... @SohanShet  Maybe replace it with some different sound effect like laser 
- Add switch to turn off sound effects

SohanShet committed:
correctSound Bug fixed

Shriteen committed:
Performance: Remove sound effect for correct input

It is causing lag of input as well as music, especiallly when consecutive words are typed quickly

Shriteen committed:
Remove unused icons

Shriteen committed:
Sound effect settings

- Add music on/off icons to repo
- Use Music icon for music and speaker icon for sounds
- Do not play sounds when sound effect turned off

Shriteen committed:
Update CREDITS

Shriteen committed:
Update CREDITS
